src/components/spreadsheet/custom-spreadsheet/custom-spreadsheet-save-dialog.tsx

```diff
@@ -37,17 +38,33 @@ export default function CustomSpreadsheetSaveDialog({ tabIndex, open }: Readonly
     }, [tabIndex, tablesDefinitionIndexes]);
 
     const customColumns = useMemo(() => {
-        return customColumnsDefinitions.map(({ id, name, formula, dependencies }) => ({
+        return customColumnsDefinitions.map(({ id, name, type, precision, formula, dependencies }) => ({
             id,
             name,
+            type,
+            precision,
             formula,
             dependencies: JSON.stringify(dependencies),
         }));
     }, [customColumnsDefinitions]);
 
-    const staticColumnIdToField = useMemo(() => {
+    const staticColumnIdToColInfos = useMemo(() => {
         const equipment = tablesDefinitionIndexes.get(tabIndex);
-        return equipment ? new Map<string, string>(equipment.columns.map((c) => [c.colId!, c.field ?? ''])) : null;
+        return equipment
+            ? new Map<string, ColumnWithFormulaDto>(
+                  equipment.columns.map((c) => [
+                      c.colId!,
+                      {
+                          id: c.field ?? '',
+                          name: c.colId!,
```

Comment:
```suggestion
                          name: c.headerComponentParams?.displayName ?? c.colId,
```
to have the correct name when we export static columns